### PR TITLE
Update documentation for v5.4.0

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
- add openssl to dependencies in RELEASE NOTES
- add link to release notes in compilation doc

Addresses #80

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ gcc        | >=4.8.x
 autoconf   | >=2.61
 zlib1g-dev | >=0.13.x
 libtool    | >=2.x
+openssl*   | >=v1.x
+
+openssl is needed if libdrizzle-redux should support
```
Comment (ben-palmer-sociomantic) [inline RELEASE_NOTES.md]: Should this line have the `*` so the one next to `openssl*` is referencing it?

**Review by ben-palmer-sociomantic — COMMENTED** [review]
Only one minor comment then LGTM

**Comment by andreas-bok-sociomantic** [comment]
Updated based on comments from @ben-palmer-sociomantic 

```diff 
diff --git a/RELEASE_NOTES.md b/RELEASE_NOTES.md
index 9d1539f..f7059f4 100644
--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -7,9 +7,9 @@ gcc        | >=4.8.x
 autoconf   | >=2.61
 zlib1g-dev | >=0.13.x
 libtool    | >=2.x
-openssl*   | >=v1.x
+openssl<sup>1</sup> | >=v1.x
 
-openssl is needed if libdrizzle-redux should support
+<sup>1</sup> openssl is needed if libdrizzle-redux is compiled with support for
 SSL connections.
 

```

**Comment by ben-palmer-sociomantic** [comment]
LGTM, happy to rebase once the commits are squashed.

**Comment by andreas-bok-sociomantic** [comment]
Squashed commits,

**Pull request merged**